File: src/Host/Controllers/Decrees/DecreeDetails.cs
using System.Collections.Generic;
using YAGO.World.Host.Controllers.Colonies.ColonyParameters;
using YAGO.World.Host.Controllers.Episodes;

namespace YAGO.World.Host.Controllers.Decrees
{
    public record DecreeDetails(
        long Id,
        string Name,
        string Image,
        string[] Text,
        IReadOnlyList<ColonyParameterResponse> Parameters,
        IReadOnlyList<ColonyParameterResponse> Requirements,
        string[] Description,
        SlideButtonResponse Button);
}

File: src/Host/ClientApp/src/entities/reforms/DecreeDetails.ts
import { apiRequester} from "../../shared/api/ApiRequester";
import type { ColonyParameter } from "../colonies/ColonyParameter";
import type { SlideButton } from "../events/Episode";

export interface DecreeDetails {
    id: number,
    name: string,
    image: string,
    text: string[],
    parameters: ColonyParameter[],
    requirements: ColonyParameter[],
    description: string[],
    button: SlideButton
}

const extendedApiSlice = apiRequester.injectEndpoints({
    endpoints: (builder) => ({
        getDecree: builder.query<DecreeDetails, number>({
            query: (id) => `decrees/getDecree?id=${id}`,
            keepUnusedDataFor: 0,
            providesTags: []
        }),
    }),
});


export const {
    useGetDecreeQuery,
} = extendedApiSlice;